ethereum/core/src/main/java/org/hyperledger/besu/ethereum/chain/DefaultBlockchain.java

```diff
@@ -127,40 +127,69 @@ private DefaultBlockchain(
 
     this.reorgLoggingThreshold = reorgLoggingThreshold;
     this.blockChoiceRule = heaviestChainBlockChoiceRule;
-    this.numberOfBlocksToCache = numberOfBlocksToCache;
 
-    if (numberOfBlocksToCache != 0) {
-      blockHeadersCache =
-          Optional.of(
-              CacheBuilder.newBuilder().recordStats().maximumSize(numberOfBlocksToCache).build());
+    initializeCaches(metricsSystem, numberOfBlockHeadersToCache, numberOfBlocksToCache);
+    createCounters(metricsSystem);
+    createGauges(metricsSystem);
+  }
+
+  private void initializeCaches(
+      final MetricsSystem metricsSystem, final int headersCacheSize, final int blocksCacheSize) {
+    if (headersCacheSize == 0 && blocksCacheSize == 0) {
+      setAllCachesEmpty();
+      return;
+    }
+
+    int headersSize =
+        (headersCacheSize != 0) ? Math.max(headersCacheSize, blocksCacheSize) : blocksCacheSize;
```

Comment:
This simplification should work the same right?
```suggestion
    final int headersSize = Math.max(headersCacheSize, blocksCacheSize);
```